Language/Haskell/GhcMod/Monad/Types.hs

```diff
@@ -129,10 +138,11 @@ data GhcModState = GhcModState {
       gmGhcSession   :: !(Maybe GmGhcSession)
     , gmComponents   :: !(Map ChComponentName (GmComponent GMCResolved (Set ModulePath)))
     , gmCompilerMode :: !CompilerMode
+    , gmMMappedFiles :: Map TargetId (StringBuffer, UTCTime)
```

Comment:
I'm not convinced using TargetId as the key is a good idea. Since a `TargetModule` might point to the same file as a `TargetFile` so it's not unique which would defeat the point of the map. You should probably just store the module name and use HomeModuleGraph to look up the corresponding file at the last minute if that's needed.